app/controllers/ileQuery/IleQueryController.scala

```diff
@@ -0,0 +1,125 @@
+  authenticate: AuthenticatedAction,
+  mcc: MessagesControllerComponents,
+  errorHandler: ErrorHandler,
+  ileQueryRepository: IleQueryRepository,
+  connector: CustomsDeclareExportsMovementsConnector,
+  ileQueryPage: ile_query,
+  loadingScreenPage: loading_screen,
+  ileQueryDucrResponsePage: ile_query_ducr_response,
+  ileQueryMucrResponsePage: ile_query_mucr_response
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport {
+
+  def displayQueryForm(): Action[AnyContent] = authenticate { implicit request =>
+    Ok(ileQueryPage(form))
+  }
+
+  def submitQueryForm(): Action[AnyContent] = authenticate { implicit request =>
+    form
+      .bindFromRequest()
+      .fold(
+        formWithErrors => BadRequest(ileQueryPage(formWithErrors)),
+        validUcr => Redirect(controllers.ileQuery.routes.IleQueryController.submitQuery(validUcr))
+      )
+  }
+
+  def submitQuery(ucr: String): Action[AnyContent] = authenticate.async { implicit request =>
+    def loadingPageResult = Ok(loadingScreenPage()).withHeaders("refresh" -> "5")
+
+    ileQueryRepository.findBySessionIdAndUcr(retrieveSessionId, ucr).flatMap {
+      case Some(query) =>
+        connector.fetchQueryNotifications(query.conversationId, request.providerId).flatMap { response =>
+          response.status match {
+            case OK =>
+              ileQueryRepository.removeByConversationId(query.conversationId).map { _ =>
+                val queryResponse = Json.parse(response.body).as[IleQueryResponse]
+
+                val ducrResult = queryResponse.queriedDucr.map(ducr => Ok(ileQueryDucrResponsePage(ducr)))
+                val mucrResult = queryResponse.queriedMucr.map(mucr => Ok(ileQueryMucrResponsePage(mucr)))
+
+                ducrResult.orElse(mucrResult).getOrElse(loadingPageResult)
+              }
+            case NO_CONTENT =>
+              Future.successful(loadingPageResult)
+            case GATEWAY_TIMEOUT =>
+              ileQueryRepository.removeByConversationId(query.conversationId).map { _ =>
+                InternalServerError(errorHandler.standardErrorTemplate())
+              }
+            case _ => Future.successful(BadRequest(errorHandler.standardErrorTemplate()))
+          }
+        }
+
+      case None => sendIleQuery(ucr)
+    }
+  }
+
+  private def sendIleQuery(ucr: String)(implicit request: AuthenticatedRequest[AnyContent]): Future[Result] =
+    form
+      .fillAndValidate(ucr)
+      .fold(
+        formWithErrors => Future.successful(BadRequest(ileQueryPage(formWithErrors))),
+        validUcr => {
+          val ileQueryRequest = buildIleQuery(request.providerId, validUcr)
+
+          connector.submit(ileQueryRequest).flatMap { conversationId =>
+            val ileQuery = IleQuery(retrieveSessionId, validUcr, conversationId)
+
+            ileQueryRepository.insert(ileQuery).map { _ =>
+              Redirect(controllers.ileQuery.routes.IleQueryController.submitQuery(ucr))
```

Comment:
why not loading screen with refresh - we would make here another request

Reply:
Because we can have a really fast response. This will allow to first check notifications before showing loading screen which can be faster

Reply:
Above approach discussed with BAs and agreed that it's better to check if there is some notifications before showing loading screen